packages/signature-controller/src/SignatureController.ts

```diff
@@ -356,6 +382,28 @@ export class SignatureController extends BaseControllerV2<
     );
   }
 
+  /**
+   * Called when the message metadata needs to be updated.
+   *
+   * @param messageId - The id of the message to update.
+   * @param metadata - The data to update the metadata property in the message.
+   */
+  setMessageMetadata(messageId: string, metadata: Json) {
+    const messageManagers = [
+      this.#messageManager,
+      this.#personalMessageManager,
+      this.#typedMessageManager,
+    ];
+
+    for (const manager of messageManagers) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      if (this.#trySetMessageMetadata(manager, messageId, metadata)) {
```

Comment:
Array of generics are always a bit awkward, does it work if we instead cast the first `manager` argument using `as any`?

Reply:
It did!